[FEATURE] Does this bpftime support arm aarch64 platform?

Does this bpftime support arm aarch64 platform?

**we met error while compiling on aarch64 platform:**

/home/nvidia/bpftrace/src/bpftime/runtime/src/bpf_helper.cpp: In function ‘void segv_read_handler(int, siginfo_t*, void*)’:
/home/nvidia/bpftrace/src/bpftime/runtime/src/bpf_helper.cpp:130:59: error: ‘struct mcontext_t’ has no member named ‘gregs’; did you mean ‘regs’?
  130 |                 auto *rip = (greg_t *)(&uctx->uc_mcontext.gregs[REG_RIP]);
      |                                                           ^~~~~
      |                                                           regs
/home/nvidia/bpftrace/src/bpftime/runtime/src/bpf_helper.cpp:130:65: error: ‘REG_RIP’ was not declared in this scope
  130 |                 auto *rip = (greg_t *)(&uctx->uc_mcontext.gregs[REG_RIP]);
      |                                                                 ^~~~~~~
/home/nvidia/bpftrace/src/bpftime/runtime/src/bpf_helper.cpp: In function ‘void segv_write_handler(int, siginfo_t*, void*)’:
/home/nvidia/bpftrace/src/bpftime/runtime/src/bpf_helper.cpp:210:59: error: ‘struct mcontext_t’ has no member named ‘gregs’; did you mean ‘regs’?
  210 |                 auto *rip = (greg_t *)(&uctx->uc_mcontext.gregs[REG_RIP]);
      |                                                           ^~~~~
      |                                                           regs
/home/nvidia/bpftrace/src/bpftime/runtime/src/bpf_helper.cpp:210:65: error: ‘REG_RIP’ was not declared in this scope
  210 |                 auto *rip = (greg_t *)(&uctx->uc_mcontext.gregs[REG_RIP]);
      |                                                                 ^~~~~~~
make[2]: *** [runtime/CMakeFiles/runtime.dir/build.make:219: runtime/CMakeFiles/runtime.dir/src/bpf_helper.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1151: runtime/CMakeFiles/runtime.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

--- a/runtime/src/bpf_helper.cpp
+++ b/runtime/src/bpf_helper.cpp
@@ -137,9 +137,15 @@ static void segv_read_handler(int sig, siginfo_t *siginfo, void *ctx)
 	} else if (status_probe_read == PROBE_STATUS::RUNNING_NO_ERROR) {
 		// set status to error
 		auto uctx = (ucontext_t *)ctx;
-		auto *rip = (greg_t *)(&uctx->uc_mcontext.gregs[REG_RIP]);
+#if defined(__x86_64__) || defined(_M_X64)
+		auto *ip = (greg_t *)(&uctx->uc_mcontext.gregs[REG_RIP]);
+#elif defined(__aarch64__) || defined(_M_ARM64)
+		auto *ip = (greg_t *)(&uctx->uc_mcontext.pc);
+#else
+#error "Unsupported architecture"
+#endif
 		status_probe_read = PROBE_STATUS::RUNNING_ERROR;
-		*rip = (greg_t)&jump_point_read;
+		*ip = (greg_t)&jump_point_read;
 	}
 }
 #endif
@@ -217,9 +223,15 @@ static void segv_write_handler(int sig, siginfo_t *siginfo, void *ctx)
 	} else if (status_probe_write == PROBE_STATUS::RUNNING_NO_ERROR) {
 		// set status to error
 		auto uctx = (ucontext_t *)ctx;
-		auto *rip = (greg_t *)(&uctx->uc_mcontext.gregs[REG_RIP]);
+#if defined(__x86_64__) || defined(_M_X64)
+		auto *ip = (greg_t *)(&uctx->uc_mcontext.gregs[REG_RIP]);
+#elif defined(__aarch64__) || defined(_M_ARM64)
+		auto *ip = (greg_t *)(&uctx->uc_mcontext.pc);
+#else
+#error "Unsupported architecture"
+#endif
 		status_probe_write = PROBE_STATUS::RUNNING_ERROR;
-		*rip = (greg_t)&jump_point_write;
+		*ip = (greg_t)&jump_point_write;
 	}
 }
 #endif

